src/main/java/frc/robot/subsystems/PhotonSubsystem.java

```diff
@@ -0,0 +1,225 @@
+      this.direction = direction;
+    }
+  }
+  private double CAMERA_HEIGHT = 0.29;
+  private Rotation2d CAMERA_PITCH = Rotation2d.fromRadians(0.305);
+  //x is forwards, y is sideways with +y being left, rotation probobly if + left too
+  private Pose2d cameraOffset = new Pose2d(new Translation2d(-0.2,0.28), Rotation2d.fromDegrees(0));
+  
+  //networkTableName
+  private String networkTableName = "PhotonCamera";
+  //data max
+  private int maxData = 10;
+  //declarations
+  private static PhotonSubsystem instance;
+  private DoubleArrayPublisher pubSetPoint;
+  private DoublePublisher pubRange, pubYaw;
+  private PhotonCamera camera1;
+  private Translation2d targetPos;
+  private Rotation2d targetRotation;
+  private LinearFilter filteryaw = LinearFilter.movingAverage(maxData);
+  private LinearFilter filterX = LinearFilter.movingAverage(maxData);
+  private LinearFilter filterY = LinearFilter.movingAverage(maxData);
+  private int data;
+  private int id;
+
+  
+
+  public static PhotonSubsystem getInstance(){
+    if (instance == null){
+      instance = new PhotonSubsystem();
+    }
+    return instance;
+  }
+
+  /** Creates a new photonAprilTag. */
+  private PhotonSubsystem() {
+    //name of camera, change if using multiple cameras
+    camera1 = new PhotonCamera("OV9281");
+    //networktable publishers
+    pubSetPoint = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleArrayTopic("PhotonAprilPoint").publish(PubSubOption.periodic(0.02));
+    pubRange = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleTopic("Range").publish(PubSubOption.periodic(0.02));
+    pubYaw = NetworkTableInstance.getDefault().getTable(networkTableName).getDoubleTopic("Yaw").publish(PubSubOption.periodic(0.02));
+    reset(-1);
+  }
+
+  public void reset(int desiredId) {
+    filterX.reset();
+    filterY.reset();
+    filteryaw.reset();
+    //set target info to the robot's info
+    targetRotation = SwerveSubsystem.getInstance().getPose().getRotation();
+    targetPos = SwerveSubsystem.getInstance().getPose().getTranslation();
+    //initialize vars
+    data = 0;
+    id = desiredId;
+  }
+
+  public Command getResetCommand(int tagid){
+    return runOnce(() -> reset(tagid));
+  }
+
+  public Command getWaitForDataCommand(int tagid){
+    return new FunctionalCommand(() -> reset(tagid), ()->{}, (interrupted) ->{}, ()->hasData(), this);
+  }
+  public Command getAprilTagCommand(PhotonPositions spacePositions){
+
+  return Commands.sequence(
+    getWaitForDataCommand(spacePositions.id),
+    new PhotonMoveToTarget(spacePositions.waypoint,0.1),
+    new PhotonMoveToTarget(spacePositions.destination, spacePositions.direction, 0.02 ));
+  }
+
+  public Translation2d getTargetPos(){
+    return targetPos;
+  }
+
+  public Rotation2d getTargetRotation(){
+    return targetRotation;
+  }
+
+  public boolean hasData() {
+    //if data is the max that the filters hold
+    return(data >= maxData);
```

Comment:
Rename data and maxData to numSamples and maxNumSamples so it's more clear for others reading your code.

And maybe it should be `NUM_SAMPLES_REQUIRED` or something similar since it's the number of samples required to take control of the robot.